### DISCUSSION TITLE
how to fix an overlapping navbar with styled components?

### DISCUSSION THREAD

**Discussion by SabSchek** [neuefische/questions#256] [Web] [answered]
Hello, 

I have a question regards styled components, as I find no way to let my nav bar looking the way I want it to look like. 
I tried out several ways to set up a grid and to put it all in the correct pattern. Unfortunately it shows the category list / navbar, but the title above disappeared. 

Maybe someone has an idea how I can style the navbar to let the title "Donatella" appear on top in with position fixed. 
Thanks in advance.
<img width="1064" alt="Bildschirmfoto 2022-10-12 um 21 29 35" src="https://user-images.githubusercontent.com/109827469/195431961-e9fb7621-379f-4f6c-95ff-ddf7d26dd36f.png">
<img width="1160" alt="Bildschirmfoto 2022-10-12 um 21 29 30" src="https://user-images.githubusercontent.com/1098274
<img width="1304" alt="Bildschirmfoto 2022-10-12 um 21 29 18" src="https://user-images.githubusercontent.com/1098274
<img width="382" alt="Bildschirmfoto 2022-10-12 um 21 23 32" src="https://user-images.githubusercontent.com/109827469/195431997-46b989ca-61af-4be1-bae0-fb3495d410ac.png">
69/195431990-031420c7-ba76-4b15-89d1-8275c39f6238.png">
<img width="1247" alt="Bildschirmfoto 2022-10-12 um 21 29 12" src="https://user-images.githubusercontent.com/109827469/195431994-398856f1-c01f-4dca-b594-2af29d3a3d11.png">
69/195431981-2a7d7076-6cdc-4ed9-a73d-411259cce25d.png">
<img width="1033" alt="Bildschirmfoto 2022-10-12 um 21 29 23" src="https://user-images.githubusercontent.com/109827469/195431988-fa87980e-28ae-498f-979e-03b5ac1424f4.png">

**Comment by NikRoe** [comment]
Hi Fine,
please send us a link a PR, so you can take a closer look.

Thanks :)

**Reply by NikRoe** [reply]
https://github.com/SabSchek/capstone-project/pull/15

**Comment by NikRoe** [comment, marked as answer]
Hi Fine,

For your `Layout` component: You need to implement it in your `_app.js` in order for it to work properly:
```js
    <>
      <GlobalStyle />
      <Layout>
        <Component {...pageProps} />
      </Layout>
    </>
```
> remember to import the component.

In your component `Layout` you are currently working with a styled component called Main, but this component is not declared anywhere. It could look like this:
```js
const Main = styled.main`
  margin-top: 100px;
`;
```

Once this component is updated, you can remove the `HeaderNav` components from the pages.

In your component `HeaderNav`, your images are missing the `alt`-tags.
Your `Headline` in this component is causing the issue with the links at the top of your navigation. 
Your `Headline` had a width of 100vw and was not assigned a place in the grid, leading to an extension of the grid. We removed the width, the position and assigned it a place within the grid:
```js
const Headline = styled.h1`
  color: #e6dacf;
  font-size: 30px;
  text-align: center;
  font-family: 'Dancing Script', cursive;
  padding-top: 5px;
  margin: 0;
  grid-column: 2/9;
`;
```

On top of that, we made a change to your `HeaderundNav` and fixed the error in the grid-template-rows property:
```js
const HeaderundNav = styled.div`
  z-index: 1;
  height: 74px;
  background-color: green;
  width: 100%;
  font-size: 16px;
  display: grid;
  grid-template-columns: 2fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 2fr;
  grid-template-rows: auto auto;
  padding-bottom: 4px;
  position: fixed;
  top: 0;
  align-content: center;
`;
```

Best,
Niklas
